database.go

```diff
@@ -0,0 +1,723 @@
+	values := TagValues(make(map[string]bool))
+
+	// see if they just want all the tag values for this key
+	if len(filters) == 0 {
+		for _, n := range names {
+			idx := t.measurements[n]
+			if idx != nil {
+				values.Union(idx.tagValues(key))
+			}
+		}
+		return values
+	}
+
+	// they have filters so just get a set of series ids matching them and then get the tag values from those
+	seriesIDs := t.SeriesIDs(names, filters)
+	return t.tagValuesForSeries(key, seriesIDs)
+}
+
+// tagValuesForSeries will return a TagValues map of all the unique tag values for a collection of series.
+func (t *database) tagValuesForSeries(key string, seriesIDs SeriesIDs) TagValues {
+	values := make(map[string]bool)
+	for _, id := range seriesIDs {
+		s := t.series[id]
+		if s == nil {
+			continue
+		}
+		if v, ok := s.Tags[key]; ok {
+			values[v] = true
+		}
+	}
+	return TagValues(values)
+}
+
+type TagValues map[string]bool
+
+// ToSlice returns a sorted slice of the TagValues
+func (t TagValues) ToSlice() []string {
+	a := make([]string, 0, len(t))
+	for v, _ := range t {
+		a = append(a, v)
+	}
+	sort.Strings(a)
+	return a
+}
+
+// Union will modify the receiver by merging in the passed in values.
+func (l TagValues) Union(r TagValues) {
+	for v, _ := range r {
+		l[v] = true
+	}
+}
+
+// Intersect will modify the receiver by keeping only the keys that exist in the passed in values
+func (l TagValues) Intersect(r TagValues) {
+	for v, _ := range l {
+		if _, ok := r[v]; !ok {
+			delete(l, v)
+		}
+	}
+}
+
+//seriesIDsForName is the same as SeriesIDs, but for a specific measurement.
+func (t *database) seriesIDsForName(name string, filters Filters) SeriesIDs {
+	idx := t.measurements[name]
+	if idx == nil {
+		return nil
+	}
+
+	// process the filters one at a time to get the list of ids they return
+	idsPerFilter := make([]SeriesIDs, len(filters), len(filters))
+	for i, filter := range filters {
+		idsPerFilter[i] = idx.seriesIDs(filter)
+	}
+
+	// collapse the set of ids
+	allIDs := idsPerFilter[0]
+	for i := 1; i < len(filters); i++ {
+		allIDs = allIDs.Intersect(idsPerFilter[i])
+	}
+
+	return allIDs
+}
+
+// MeasurementBySeriesID returns the Measurement that is the parent of the given series id.
+func (t *database) MeasurementBySeriesID(id uint32) *Measurement {
+	return t.seriesToMeasurement[id]
+}
+
+// MeasurementAndSeries returns the Measurement and the Series for a given measurement name and tag set.
+func (t *database) MeasurementAndSeries(name string, tags map[string]string) (*Measurement, *Series) {
+	idx := t.measurements[name]
+	if idx == nil {
+		return nil, nil
+	}
+	return idx, idx.seriesByTags(tags)
+}
+
+// SereiesByID returns the Series that has the given id.
+func (t *database) SeriesByID(id uint32) *Series {
+	return t.series[id]
+}
+
+// Measurements returns all measurements that match the given filters.
+func (t *database) Measurements(filters []*Filter) []*Measurement {
```

Comment:
I don't see any use of the `filters` in this function. Is it deliberately incomplete?